### ISSUE TITLE
[Bug]: Desktop app is unable to login using gitlab

### ISSUE THREAD

**Issue by ageekhere** [mattermost/desktop#2740]
### Before you file a bug report

- [X] I have checked the [issue tracker](https://github.com/mattermost/desktop/issues) and have not found an issue that matches the one I'm filing.
- [X] This issue is not a troubleshooting question. Troubleshooting questions go here: https://forum.mattermost.com/c/trouble-shoot/16.
- [X] This issue is not a feature request. You can request features and make product suggestions here: https://mattermost.com/suggestions/.
- [X] This issue doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/mattermost-server/issues).
- [X] This issue reproduces on the most recent [stable version](https://github.com/mattermost/desktop/releases/latest), or the most recent [prerelease version](https://github.com/mattermost/desktop/releases) of the Mattermost Desktop App.
- [X] I have read the [contribution guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

### Mattermost Desktop Version

5.3.1

### Operating System

Windows 11 x64

### Mattermost Server Version

Mattermost Version: 7.9.1 Database Schema Version: 104 Database: postgres

### Steps to reproduce

1. Be logged out of GitLab in Mattermost
2. Open Mattermost desktop app
3. Click the GitLab login button
4. White screen comes up and cannot log in via GitLab

### Expected behavior

When the user is logged out of GitLab Mattermost and they open Mattermost desktop app and click the GitLab login button you should be directed to the GitLab login page so the user can log into Mattermost via a GitLab user.

### Observed behavior

A white screen appears and the use is unable to log in.

### Log Output

```shell
[2023-05-27 18:34:47.632] [debug] DownloadsManager.init { 'this.json': {}, validatedJSON: {} }
[2023-05-27 18:34:47.638] [debug] DownloadsManager.saveAll
[2023-05-27 18:34:47.640] [debug] DownloadsManager.checkForDeletedFiles
[2023-05-27 18:34:49.273] [warn]  couldn't show serverName Chat___TAB_MESSAGING, not ready
[2023-05-27 18:34:49.307] [warn]  Failed to set new language en-US
[2023-05-27 18:34:49.308] [warn]  Failed to set new language AU
[2023-05-27 18:34:49.366] [info]  config.autostart has been configured: true
[2023-05-27 18:34:49.594] [info]  Can't send reload-config, will retry
[2023-05-27 18:34:49.625] [info]  config.autostart has been configured: true
[2023-05-27 18:34:49.722] [info]  Can't send maximized_change, will retry
[2023-05-27 18:34:52.138] [info]  hide back button
[2023-05-27 18:34:52.215] [info]  hide back button
[2023-05-27 18:34:52.289] [info]  hide back button
[2023-05-27 18:34:54.200] [info]  Checking for update
[2023-05-27 18:34:54.327] [info]  serverName Chat___TAB_MESSAGING timeout expired will show the browserview
[2023-05-27 18:34:54.420] [info]  serverName Chat___TAB_FOCALBOARD timeout expired will show the browserview
[2023-05-27 18:34:54.466] [info]  serverName Chat___TAB_PLAYBOOKS timeout expired will show the browserview
[2023-05-27 18:34:55.685] [warn]  Ignoring certificate for unmatched origin https://serverName.com:8050, will not trust
[2023-05-27 18:34:57.127] [info]  Update for version 5.3.1 is not available (latest version: 5.3.1, downgrade is disallowed).
```

### Additional Information

This works when using chrome and older versions of the desktop app.
If you install an older version such as 5.1 then log in via GitLab, uninstall 5.1 and reinstall 5.3.1 because you are already signed in when you open Mattermost desktop app you can get pass the GitLab login.

**Comment by devinbinnie** [comment]
I'm guessing it's related to a certificate mismatch:
`[2023-05-27 18:34:55.685] [warn]  Ignoring certificate for unmatched origin https://serverName.com:8050, will not trust`

We're planning on re-working the sign-in flow for servers to avoid having users run into these issues.

**Comment by ageekhere** [comment]
Did something change from 5.1 to 5.3 to affect this?

**Comment by devinbinnie** [comment]
Likely yes, we've tightened certain security to stop external sites from working in the Desktop App (since they shouldn't be able to), but it might have broken your auth flow.

**Comment by gregorpirgie** [comment]
Me and my colleagues have the same issue; it would be great if you could give this more attention. It makes the desktop app completely unusable at the moment.

**Comment by ageekhere** [comment]
> It does not work with Mattermost 5.4.0 as well

What Mattermost server Version are you running? I am on 7.10.2 and with using Mattermost desktop app 5.4.0 it seems that this issue could now be fixed.

Can others confirm that the issue is fixed when updating Mattermost server to 7.10.2 with using Mattermost desktop app 5.4.0.

**Comment by martinantonmueller** [comment]
> > It does not work with Mattermost 5.4.0 as well
> 
> What Mattermost server Version are you running? I am on 7.10.2 and with using Mattermost desktop app 5.4.0 it seems that this issue could now be fixed.
> 
> Can others confirm that the issue is fixed when updating Mattermost server to 7.10.2 with using Mattermost desktop app 5.4.0.



> What Mattermost server Version are you running? I am on 7.10.2 and with using Mattermost desktop app 5.4.0 it seems that this issue could now be fixed.

We're on
`Mattermost Version: 7.10.3
Database Schema Version: 108
Database: postgres`

**Comment by martinantonmueller** [comment]
The server is on the always on "latest", so it updates itself. Looking further into the issue it appears that the time-limit of the standalone-app is too short: https://challenges.cloudflare.com/ is used by gitlab to check for human activity. The main cause is the timeout set in the Mattermost Desktop App. When I copy the link from the Mattermost Desktop Client log and paste it into the browser, I finally get the Gitlab login page after 10 seconds. The Mattermost Desktop Client app only returns the error "handleSessionExpired false". 

So is there a way to give the desktop app more time?

**Comment by gregorpirgie** [comment]
Hey, this still does not work. Still can't connect to gitlab via the desktop app and therefore it is completely unusable.

**Comment by romen** [comment]
Hi, is there any update on this?

**Comment by gregorpirgie** [comment]
Can confirm that this is still an issue. Any updates would be very welcome.

**Comment by romen** [comment]
I think @martinantonmueller was onto the right path with his comment: https://github.com/mattermost/desktop/issues/2740#issuecomment-1614244099

@devinbinnie I tried to figure out which timeout is too short but I don't know the internals of the desktop app enough to navigate. Would you be able to point to the likely place for this so that I could test and propose a PR to solve this sooner rather than later?

**Comment by romen** [comment]
BTW I can confirm this still happens with the latest nightly build (20230927)

**Comment by devinbinnie** [comment]
Hey all,

I want to apologize for the lack of communication on these tickets regarding authentication. But I come with some good news: we've been working on separating the external login flow from the Desktop App, deferring to the browser like many other applications do. This should overall improve security and stability around the login process, as the current implementation requires a few hacks to make it work correctly. 

As of today, we released Mattermost v9.1 which has this feature implemented, and should work with all existing Desktop App versions going back to at least v5.4.0. Going forward we will be supporting this login flow for all external providers using the Desktop App, and this should fix any issues around login flow. Your server will need to be upgraded to take advantage of this feature.

I'll be closing these tickets for now as fixed, but feel free to comment and ask questions if you have any. Thanks for your patience :)

**Comment by acdh-ch** [comment]
Thank you! Yes, now it uses default web browser for login and login works.
